a10_octavia/controller/worker/tasks/vthunder_tasks.py

```diff
@@ -296,45 +298,111 @@ def execute(self, vthunder, loadbalancer, added_ports, ifnum_master=None, ifnum_
-                        if interfaces['interface']['ethernet-list'][i]['action'] == "disable":
-                            ifnum = interfaces['interface']['ethernet-list'][i]['ifnum']
+                        ifnum = interfaces['interface']['ethernet-list'][i]['ifnum']
+                        if subnet.ip_version == 6:
+                            if not ifnum_address[ifnum]:
+                                LOG.warning("IPv6 Address for ethernet %s not found in configurations", ifnum)
+                                continue
+                            if backup_vthunder:
+                                self.axapi_client.device_context.switch(2, None)
+                                self.axapi_client.system.action.setInterface(ifnum, ifnum_address[ifnum],
+                                                                             subnet.ip_version)
+                            else:
+                                self.axapi_client.device_context.switch(1, None)
+                                self.axapi_client.system.action.setInterface(ifnum, ifnum_address[ifnum],
+                                                                             subnet.ip_version)
+                        elif interfaces['interface']['ethernet-list'][i]['action'] == "disable":
                             self.axapi_client.device_context.switch(1, None)
-                            self.axapi_client.system.action.setInterface(ifnum)
+                            self.axapi_client.system.action.setInterface(ifnum, None,
+                                                                         subnet.ip_version)
                             self.axapi_client.device_context.switch(2, None)
-                            self.axapi_client.system.action.setInterface(ifnum)
+                            self.axapi_client.system.action.setInterface(ifnum, None,
+                                                                         subnet.ip_version)
         except(acos_errors.ACOSException, req_exceptions.ConnectionError) as e:
             LOG.exception("Failed to configure ethernet interface vThunder: %s", str(e))
             raise e
 
 
+class GetValidIPv6Address(VThunderBaseTask):
+
+    @axapi_client_decorator
+    def execute(self, loadbalancer, vthunder, subnet, loadbalancers_list):
+        ipv6_address_list = {}
+        if subnet.ip_version != 6:
+            return None
+        topology = CONF.a10_controller_worker.loadbalancer_topology
+        amphora_id = loadbalancer.amphorae[0].id
+        compute_id = loadbalancer.amphorae[0].compute_id
+        network_driver = utils.get_network_driver()
+        nics = network_driver.get_plugged_networks(compute_id)
+        if topology == "ACTIVE_STANDBY":
+            backup_nics = network_driver.get_plugged_networks(loadbalancer.amphorae[1].compute_id)
+            nics = nics + backup_nics
+        if subnet.ip_version == 6:
```

Comment:
this should always be true since we check this in line 342

Reply:
DONE